### DISCUSSION TITLE
Return multiple passes

### DISCUSSION THREAD

**Discussion by brannan-holland** [alexandercerutti/passkit-generator#222] [Q&A] [answered]
Thank you @alexandercerutti for putting this well documented package together!

I have been attempting to implement an "Add all" option for passes that are grouped together with little success.

Two key issues I have encountered:

- Only way I could figure to add pass together was to concat the buffer for each. This returned a zip file. Im sure there is a much better way to do this.
- I have not found a great way to modify the "groupingIdentifier" when using form method.

New to a lot of this and do not post often. Please let me know if there is anything else I can include!

**Comment by alexandercerutti** [comment, marked as answer]
Hey @brannan-holland! Thanks for using passkit-generator!

Returning multiple passes is a weird matter.
Apple officially supports `.pkpasses`, which is a bundle containing multiple `.pkpass` inside, but apparently only a few apps support them: Safari, Mail and I don't know who else, and if there is something that developers can do in their app to support them.

passkit-generator has the support to `.pkpasses` if you need them.

Otherwise you'll need to implement something custom I guess, which involves designing how you are receiving your .pkpass files in the app (because that means you cannot probably use another way)

For what concerns the `groupingIdentifier`, I'm not exactly sure what you are asking.

Let me know!

**Reply by brannan-holland** [reply]
Ill look into .pkpasses and see if this is a good option!

I have the `groupingIdentifier` configured in my .pass model currently. I know I will need to dynamically set this value as im generating the passes. I see I can set certain fields such as barcode or text fields using `PKPass.from`. I cant seem to find the best way to set `groupingIdentifier`

**Reply by alexandercerutti** [reply]
@brannan-holland `groupingIdentifier` is not a property that requires a strict validation on passkit-generator side. For this reason, you can pass it in the props when you initialize `PKPass.from`.

Look here: https://github.com/alexandercerutti/passkit-generator/wiki/API-Documentation-Reference#-from-a-template

Look in the page for `getAdditionalPropsForThisPassSomehow()` example function to see how it gets used.

https://github.com/alexandercerutti/passkit-generator/blob/bab4117cb029b00a50077262187fcc2ad9412357/examples/self-hosted/src/PKPasses.ts#L57-L77

`new PKPass` is not different from `PKPass.from` in this sense.

The code you see above, is from an example in the repo that shows how to create `.pkpasses`. I suggest you to fully read it, from top to bottom comments included to understand more.

**Comment by brannan-holland** [comment]
Disregard. .pkpasses looks to solve both problems!

Thank you for your help!
